### ISSUE TITLE
Not Authorized to access getSomeObject on type Query when result is empty.

### ISSUE THREAD

**Issue by lenarmazitov** [aws-amplify/amplify-cli#4907]
**Describe the bug**
Hello, seems like something changed in amplify or appsync not so long time ago.
When I try to perform GraphQL query which returns **empty** result, now I have error:
```
Not Authorized to access getSomeObject on type Query 
```
There is code in resolver which leads to this behavior:
```
#if( !($isStaticGroupAuthorized == true || $isDynamicGroupAuthorized == true || $isOwnerAuthorized == true) )
  $util.unauthorized()
#end
```
Thats right code, but somehow previously when **$ctx.result** was empty I did not get this error.

**The problem is that Apollo don't cache query because error occurred.**

This is wrong behavior, because if **$ctx.result** is NULL there should not be error

**To Reproduce**
1. amplify add auth
2. amplify add api with any schema with
```
@auth(rules: [{allow: owner, ownerField: "owner"}])
```
3. authenticate user
4. execute query getSomething(id) on where sure no data exists
5. See error

**Expected behavior**
On empty result error is not necessary because no data returned.

**Using**
- amplify-cli@4.24.3 
- react-native

**Comment by RayStroud** [comment]
We're experiencing the same behavior after upgrading to 4.24.3 from 4.22.0. Reverting to 4.24.2 didn't work for us.

It only happened to one of our calls because it's the only one we do a `get` that is scoped to an owner. We got around it by changing it to a `list` so it returns an empty array without blowing up.

**Comment by zetsnotdead** [comment]
We are facing the same issue after updating from 4.24.1 to 4.25.0

**Comment by simon-lanf** [comment]
We are experiencing this problem too. Some nested model properties do not work in the cloud when using the latest version. Pinning 4.24.1 in the Amplify Console fixed our issue. Please fix this otherwise we will never update.

**Comment by jwcarroll** [comment]
Just ran into this issue as well and it basically broke production for me. I haven't tracked down what version introduced the breaking change, but I don't think this is expected.

Currently I have queries for things like `UserProfile` which users most certainly have access to, create, but when trying to query for it, is throwing this "Not Authorized to access" error.

**Comment by jwcarroll** [comment]
Just as an update, this appears to be fixed as of `4.27.3`

I don't know what version exactly this was fixed in, but a small change to the VTL templates can be seen now:

```
#if( $util.isNullOrEmpty($ctx.result) )
  #return
#end
```

Which... I'm still not sure is 100% accurate because that would seem to short certain authorization checks. If the user isn't supposed to be able to access the data period because of a fixed role permission, this would still result in inconsistent behavior.

**Comment by dylan-westbury** [comment]
```
  @model
  @searchable
    { allow: groups, groups: ["Admins"] }
    { allow: owner, operations: [create, update] }
  ])
```

List items is fine, but get item throws "not authorized" (for none-owner)

amplify -v `4.34.0`

**Comment by dylan-westbury** [comment]
I was receiving this error  "Not Authorized to access getSomeObject on type Query", I resolved by adding the group of the user making query.

**From:**
```
  @model
  @searchable
    { allow: groups, groups: ["Admins"] }
    { allow: owner, operations: [create, update] }
  ])
```

**To**
```
  @model
  @searchable
    { allow: groups, groups: ["Admins"] }
    { allow: groups, groups: ["Customers"], operations: [read] }
    { allow: owner, operations: [create, update] }
  ])
```

**Comment by kaustavghosh06** [comment]
Based on @jwcarroll's comment - this was fixed with v 4.27.3 and we haven't see any reports of this issue post that. Closing this issue. Please let us know if you hit into this issue and we can re-open.
